flaky TestProvidesExpire test

```
=== RUN   TestProvidesExpire
      providers_manager_test.go:218: expected everything to be cleaned out of the datastore
  --- FAIL: TestProvidesExpire (1.15s)

```
=== RUN   TestProvidesExpire
    providers_manager_test.go:210: expected providers to still be there
--- FAIL: TestProvidesExpire (0.86s)
```

https://github.com/libp2p/go-libp2p-kad-dht/actions/runs/13944782424/job/39029368896?pr=1067#step:14:311

```
=== RUN   TestProvidesExpire
    providers_manager_test.go:210: expected providers to still be there
--- FAIL: TestProvidesExpire (0.78s)
```

https://github.com/libp2p/go-libp2p-kad-dht/actions/runs/14735084702/job/41359046360?pr=1079#step:13:3495

--- a/providers/providers_manager_test.go
+++ b/providers/providers_manager_test.go
@@ -195,7 +195,11 @@ func TestProvidesExpire(t *testing.T) {
 		}
 	}
 
-	time.Sleep(ProvideValidity/2 + 2*defaultCleanupInterval)
+	// Sleep long enough for first batch to expire, but not the second batch
+	// First batch: added at t=0, expires at t=500ms (ProvideValidity)
+	// Second batch: added at t=250ms, expires at t=750ms
+	// We sleep 350ms (total 600ms) to provide 150ms margin before second batch expires
+	time.Sleep(ProvideValidity/2 + defaultCleanupInterval)
 
 	for _, h := range mhs[:5] {
 		out, _ := p.GetProviders(ctx, h)

--- a/records/providers_manager_test.go
+++ b/records/providers_manager_test.go
@@ -195,7 +195,11 @@ func TestProvidesExpire(t *testing.T) {
 		}
 	}
 
-	time.Sleep(ProvideValidity/2 + 2*defaultCleanupInterval)
+	// Sleep long enough for first batch to expire, but not the second batch
+	// First batch: added at t=0, expires at t=500ms (ProvideValidity)
+	// Second batch: added at t=250ms, expires at t=750ms
+	// We sleep 350ms (total 600ms) to provide 150ms margin before second batch expires
+	time.Sleep(ProvideValidity/2 + defaultCleanupInterval)
 
 	for _, h := range mhs[:5] {
 		out, _ := p.GetProviders(ctx, h)

